dhall/src/Dhall.hs

```diff
@@ -731,9 +779,11 @@ bool :: Decoder Bool
 bool = Decoder {..}
   where
     extract (BoolLit b) = pure b
-    extract expr        = typeError expected expr
+    extract expr        = typeError expected' expr
```

Comment:
It seems like it would require less code to replace all uses of `typeError` with `typeError'`, like this:

```haskell
extract (BoolLit b) = pure b
extract  expr       = typeError' expected expr

expected = pure Bool
```

Reply:
I got rid of `typeError'` and it looks much nicer.